### PR TITLE
Cobra cli refactor

### PR DESCRIPTION
**Pull request by prnvbn**
Still need to delete the urfave/cli stuff.

### CODE REVIEW

File: cmd/all.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+var allCmd = &cobra.Command{
+	Use:   "all",
+	Short: "download all files from scientia",
+	Long:  `TODO`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		for _, course := range courses {
+			if course.HasMaterials {
+				err := downloadCourse(course, newOnly)
+				if err != nil {
+					fmt.Println(err)
```
Comment (Akshat-Tripathi) [inline cmd/all.go:19]: Maybe instead of printing them we spit out a list of docs that we couldn't download

File: cmd/course.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/*
+Copyright © 2022 NAME HERE <EMAIL ADDRESS>
+
```
Comment (Akshat-Tripathi) [inline cmd/course.go]: Copyright?

File: cmd/download.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	if err != nil {
+		return err
+	}
+	dirName := course.FullName()
+	saveDir := path.Join(cfg.SaveDir, dirName)
+	err = os.Mkdir(saveDir, 0777)
+
+	if err != nil && !errors.Is(err, os.ErrExist) {
+		if errors.Is(err, os.ErrNotExist) {
+			log.Warnf("Directory %s does not exist", saveDir)
```
Comment (Akshat-Tripathi) [inline cmd/download.go:61]: But didn't you just make it?

**Review by Akshat-Tripathi — APPROVED** [review]
Lgtm but there are a few odd bits

File: cmd/course.go (hunk 2 above)
Comment (prnvbn) [reply cmd/course.go]: ahh yes it is auto-generated by cobra-cli, will get rid of it

File: cmd/all.go (hunk 1 above)
Comment (prnvbn) [reply cmd/all.go:19]: yeahh, I think let's track it in another PR. We need to make the output of the command more useful

File: cmd/download.go (hunk 3 above)
Comment (prnvbn) [reply cmd/download.go:61]: yeah, forgot to delete the outer mkdir

File: cmd/all.go (hunk 1 above)
Comment (Akshat-Tripathi) [reply cmd/all.go:19]: Yep that makes sense to me

**Pull request merged**